app/models/web_results_post_processor.rb

```diff
@@ -1,9 +1,10 @@
-class WebResultsPostProcessor
+class WebResultsPostProcessor < PostProcessor
```

Comment:
Creating a `PostProcessor` class isn't necessarily a bad idea. (However, I would give it a less generic name such as `SearchResultsPostProcessor`. It actually seems like `WebResultsPostProcessor` would be a logical parent class for any of the search types that returns standard url/description/title results, but I haven't looked deeply into the logical division between those classes.) There's a faction of Ruby developers who argue that there should never be "abstract" super classes that will never be instantiated directly (I'm not in that faction).

The other way to pet that cat is to create, say, a `SearchResultsPostProcessor` module that you can then mix into the other processor classes. A perfect example of this pattern is the `ResultsRejector` module, which is `include`d in a couple of the post processor classes:
```
git grep ResultsRejector
app/models/i14y_post_processor.rb:  include ResultsRejector
app/models/web_results_post_processor.rb:  include ResultsRejector
lib/results_rejector.rb:module ResultsRejector
```

Reply:
I considered something like this, but it seems not strictly needed at the moment. But that may change as we develop more of the new SERP, so I'll keep this in mind.